### PR TITLE
apimachinery: misc unstructured tests

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by sttts** [comment]
@ash2k can you take a look? The tests are incomplete, but better than nothing.

File: staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	}
+	for _, tt := range tests {
+		t.Run(tt, func(t *testing.T) {
+			js := jsonDecodeOrDie(t, tt)
+			clone := deepCopyJSON(js)
+			bs, err := jsoniter.Marshal(clone)
+			if err != nil {
+				t.Fatalf("Failed to marshal %#v: %v", clone, err)
+			}
+			if !reflect.DeepEqual(tt, string(bs)) {
```
Comment (ash2k) [inline staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go]: Just compare strings directly? `tt != string(bs)`

**Review by ash2k — APPROVED** [review]

**Comment by ash2k** [comment]
Are there fuzzing tests for types where deep copy methods are tested? `Unstructured` could be tested that way too.

File: staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go]: indeed :)

**Comment by sttts** [comment]
@ash2k fuzzer tests do not really test the machinery around. I trust the deepcopy (it's a simple recursion with a handful of cases). But I don't trust that all code around it creates a JSON like data structure. I.e. we saw *int64 before, I found uint64 now. We might have more of those bugs. On the other hand, we should not blindly add types to deepcopy. The panic in deepcopy is only a sympton, we should not create non-JSON data in the first place.

File: staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go]: fixed

**Comment by sttts** [comment]
> Unstructured could be tested that way too.

We could fuzz apimachinery types and run ToUnstructured on them.

**Pull request closed without merging**